firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -637,6 +705,29 @@ public void testDoesNotRetryOnPermanentError() {
     assertEquals(1, count.get());
   }
 
+  @Test
+  public void testRetryOnAlreadyExistsError() {
+    final FirebaseFirestore firestore = testFirestore();
+    DocumentReference doc = firestore.collection("foo").document();
+    CountDownLatch latch = new CountDownLatch(2);
```

Comment:
Use `AtomicInteger` instead of `CountDownLatch`.

`CountDownLatch` is used to coordinate the order of execution of two different threads. But you are just using as a counter, so `AtomicInteger` is what you want.